connectors/src/connectors/microsoft/temporal/file.ts

```diff
@@ -163,50 +151,26 @@ export async function syncOneFile({
     );
   }
 
-  async function getDocumentContent() {
-    try {
-      const converted = await mammoth.convertToHtml({
-        buffer: Buffer.from(downloadRes.data),
-      });
-
-      const extracted = new turndown()
-        .remove(["style", "script", "iframe", "noscript", "form", "img"])
-        .turndown(converted.value);
-
-      return extracted.trim();
-    } catch (err) {
-      localLogger.error(
-        {
-          error: err,
-        },
-        `Error while converting docx document to text`
-      );
-      throw err;
-    }
-  }
-
   let documentSection: CoreAPIDataSourceDocumentSection | null = null;
   if (
-    mimeType ===
-    "application/vnd.openxmlformats-officedocument.presentationml.presentation"
+    [
+      "application/vnd.openxmlformats-officedocument.presentationml.presentation",
+      "application/pdf",
+      "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
+    ].includes(mimeType)
   ) {
     const data = Buffer.from(downloadRes.data);
-    documentSection = await handlePptxFile(
-      data,
-      file.id,
-      localLogger,
-      maxDocumentLen
-    );
+    documentSection = await handleTextExtraction(data, localLogger, mimeType);
   } else if (mimeType === "application/vnd.ms-excel") {
     const data = Buffer.from(downloadRes.data);
-    const isSuccessful = await handleCsvFile(
+    const isSuccessful = await handleCsvFile({
```

Comment:
does handleCsvFile support vnd.ms-excel .. ? shouldn't it be text/csv ?

Reply:
Indeed, csv files are identified as "vnd.ms-excel" ... However if you upload a legacy xls file ( https://file-examples.com/index.php/sample-documents-download/sample-xls-download/ ), it will also have the vnd.ms-excel but won't be parseable, which make the activity crash endlessly. We should catch the error and ignore if it's not parseable ( just move parseAndStringifyCsv inside the try/catch)
And maybe, still test `mimeType === "application/vnd.ms-excel" || mimeType === "text/csv"` , just in case ..?